### ISSUE TITLE
OAuth 2.0 Client Credentials flow does not encode client secret

### ISSUE THREAD

**Issue by paulius-valiunas** [ArchGPT/insomnium#137]
### Expected Behavior

As described in [RFC6749](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1), both the client id and client secret values should be:
> encoded using the "application/x-www-form-urlencoded" encoding algorithm per [Appendix B](https://datatracker.ietf.org/doc/html/rfc6749#appendix-B)

Insomnium should either encode them automatically, or at least provide a button in the UI to do it manually so I don't have to use external tools.

### Actual Behavior

Insomnium skips the URL encoding step and concatenates these values exactly as they are in the UI text fields. 

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/ArchGPT/insomnium/issues) for this problem.

### Insomnium Version

0.2.3-a

### What operating system are you using?

Windows

### Operating System Version

Windows 11 version 23H2

### Installation method

winget

**Comment by archywillhe** [comment]
hey thanks; do you know what's the status at Kong/Insomnium's on this? Doesn't look like they follow the proposed standard too and this appears to break many existing configs users had

**Comment by paulius-valiunas** [comment]
Yeah I'm pretty sure they have the same problem. Do you think we should offer the user a choice whether to encode the credentials or not? I'll need some help with the UI for this, because I'm like a 100% backend developer 😅 but a simple checkbox might work.

On the other hand, sometimes you have to re-break what's broken to fix it. It's up to you.
